tests/test_more.py

```diff
@@ -1690,6 +1691,22 @@ def test_negative_repeat(self):
         expected = []
         self.assertEqual(actual, expected)
 
+    def test_infinite_input(self):
+        repeater = mi.repeat_each(cycle('AB'))
+        actual = []
+        counter = 0
+
+        while True:
+            current = next(repeater)
+            actual.append(current)
+            counter += 1
+
+            if counter == 6:
+                break
+
```

Comment:
I think this whole block can just be `mi.take(6, repeater)` ?

Reply:
Yes, indeed. I have no idea where my head was at.